extras/indexing/src/main/java/mvm/rya/indexing/external/PrecomputedJoinIndexer.java

```diff
@@ -0,0 +1,255 @@
+        @Override
+        public Configuration get() {
+            return getConf();
+        }
+    };
+
+    /**
+     * Provides access to the Accumulo {@link Connector} that was provided to
+     * this class using {@link #setConnector(Connector)}.
+     */
+    private final Supplier<Connector> accumuloSupplier = new Supplier<Connector>() {
+        @Override
+        public Connector get() {
+            return accumuloConn.get();
+        }
+    };
+
+    /**
+     * Creates and grants access to the {@link PrecomputedJoinStorage} that will be used
+     * to interact with the PCJ results that are stored and used by Rya.
+     */
+    private final PrecomputedJoinStorageSupplier pcjStorageSupplier =
+            new PrecomputedJoinStorageSupplier(
+                    configSupplier,
+                    new AccumuloPcjStorageSupplier(configSupplier, accumuloSupplier));
+
+    /**
+     * Creates and grants access to the {@link PrecomputedJoinUpdater} that will
+     * be used to update the state stored within the PCJ tables that are stored
+     * in Accumulo.
+     */
+    private final PrecomputedJoinUpdaterSupplier pcjUpdaterSupplier =
+            new PrecomputedJoinUpdaterSupplier(
+                    configSupplier,
+                    new FluoPcjUpdaterSupplier(configSupplier));
+
+    @Override
+    public void setConf(final Configuration conf) {
+        this.conf = Optional.fromNullable(conf);
+    }
+
+    @Override
+    public Configuration getConf() {
+        return this.conf.get();
+    }
+
+    /**
+     * Set the connector that will be used by {@link AccumuloPcjStorage} if the
+     * application is configured to store the PCJs within Accumulo.
+     */
+    @Override
+    public void setConnector(final Connector connector) {
+        checkNotNull(connector);
+        accumuloConn = Optional.of( connector );
+    }
+
+    /**
+     * This is invoked when the host {@link RyaDAO#init()} method is invoked.
+     */
+    @Override
+    public void init() {
+        pcjStorageSupplier.get();
+        pcjUpdaterSupplier.get();
+    }
+
+    @Override
+    public void storeStatement(final RyaStatement statement) throws IOException {
+        checkNotNull(statement);
+        storeStatements( Collections.singleton(statement) );
+    }
+
+    @Override
+    public void storeStatements(final Collection<RyaStatement> statements) throws IOException {
+        checkNotNull(statements);
+        try {
+            pcjUpdaterSupplier.get().addStatements(statements);
+        } catch (final PcjUpdateException e) {
+            throw new IOException("Could not update the PCJs by adding the provided statements.", e);
+        }
+    }
+
+    @Override
+    public void deleteStatement(final RyaStatement statement) throws IOException {
+        checkNotNull(statement);
+        try {
+            pcjUpdaterSupplier.get().deleteStatements( Collections.singleton(statement) );
```

Comment:
If you provide a Fluo PcjUpdater, then it just logs that delete isn't supported. In theory another implementation could support deletes (like, maybe this triggers a batch job), so it's appropriate for the indexer to pass the delete command along to the updater.